sdk/storage/azure-storage-common/src/storage_per_retry_policy.cpp

```diff
@@ -28,6 +29,25 @@ namespace Azure { namespace Storage { namespace _internal {
               .ToString(Azure::DateTime::DateFormat::Rfc1123));
     }
 
+    const char* HttpHeaderTimeout = "timeout";
+    auto cancelTimepoint = ctx.CancelWhen();
+    // precison loss during time point conversion
+    if (std::chrono::duration_cast<std::chrono::microseconds>(
+            Azure::Core::Context::time_point::max() - cancelTimepoint)
+            .count()
+        <= 1000)
+    {
+      request.GetUrl().RemoveQueryParameter(HttpHeaderTimeout);
```

Comment:
nit. Add a comment about nothing would happen (no throw) if the query is not there.